### DISCUSSION TITLE
[Style] Crates folder structure

### DISCUSSION THREAD

**Discussion by DmitryDodzin** [metalbear-co/mirrord#447] [Polls]
After browsing the linkerd repo I've noticed a nice organization structure and I suggest a small change to the folder structure of the rust crates in our project with inspiration from linkerd

```
- mirrord
  - agent // crate name: mirrord-agent
  - auth // crate name: mirrord-auth
  - cli // crate name: mirrord-cli
  - config // crate name: mirrord-config (from branch #409 )
    - derive // crate name: mirrord-config-derive (probably will be in branch #409 )
  - layer // crate name: mirrord-layer
    - macro // crate name: mirrord-layer-macro (from previous mirrord-macro)
  - protocol // crate name: mirrord-protocol
- sample
  - rust // crate name: sample-rust
- tests // crate name: tests
```

For example this can help with image building
A) workspace rules can be updated to include `mirrord/*`  and no need to update `Cargo.toml` for each new package 
B) no need to include all the workspace into mirrord-agent image builder phase so cached docker layers should work when no changes affect the crates of mirrord-agent

**Comment by eyalb181** [comment]
Looks like the motion passes!
